R/drive_upload.R

```diff
@@ -66,35 +66,49 @@ drive_upload <- function(file = NULL,
   path <- path %||% root_folder()
   path <- as_dribble(path)
   confirm_single_file(path)
+  existing <- NULL
   if (!is_folder(path)) {
-    stop(
-      glue(
-        "Requested parent folder does not exist:\n{path$name}"
-      ),
-      call. = FALSE
-    )
+    if (!is.null(name)) {
+      stop(
+        glue(
+          "Requested parent folder does not exist:\n{path$name}"
+        ),
+        call. = FALSE
+      )
+    }
+    if (!overwrite) {
+      stop(
+        glue("File already exists:\n{path$name}"),
+        call. = FALSE
+      )
+    }
+    existing <- path
+    path <- unlist(promote(path, "parents")$parents)
```

Comment:
You also need to worry about parents having length greater than 1. But I actually think we shouldn't be monkey with the parents if an existing non-folder file target anyway. I'm about to make some higher level comments.